correct_periscope_drift.py

```diff
@@ -0,0 +1,430 @@
+
+    # We close the parameter file here; if you need to write
+    # values to the file you can either leave it open or -
+    # possibly better - is to close it and re-open it later.
+    #
+    pio.paramclose(fp)
+
+    # Set tool and module verbosity
+    set_verbosity(verbose)
+
+    # check outfile, exiting if it exists and clobber is
+    # not set. Note: the error message should be updated to better
+    # match that used by CIAO tools but I have not done this.
+    #
+    #
+    if not clobber and os.path.exists(mypars['corr_asolfile']):
+        raise IOError("clobber is no and outfile ({0}) exists".format(
+                mypars['corr_asolfile']))
+
+    mypars['clobber'] = clobber
+    mypars['verbose'] = verbose
+
+    # Return a dictionary with useful info (you could return a tuple or
+    # something else, depending on the parameter file)
+    #
+    return mypars
+
+
+# Display parameter info to the user.
+# The format and choice of information to display is up to you;
+# I've chosen to include some ancilalry information but it
+# depends on the tool what should be used.
+#
+def display_start_info(opts):
+    v1("Running: {0}".format(opts["progname"]))
+    v2("  version = {0}".format(VERSION))
+    v2("with parameters:")
+    v2("  evtfile={0}".format(opts["evtfile"]))
+    v2("  input_asolfile={0}".format(opts["input_asolfile"]))
+    # probably other values here too
+    v2("  verbose={0}".format(opts["verbose"]))
+    v2("  and CALDB is set to  {0}".format(os.environ["CALDB"]))
+    v2("  and ASCDS_INSTALL is {0}".format(os.environ["ASCDS_INSTALL"]))
+    v2("-" * 60)
+
+
+def equatorial2transform(ra, dec, roll):
+    """Construct the transform/rotation matrix from RA,Dec,Roll (given in degrees)
+    :returns: transform matrix
+    :rtype: Nx3x3 numpy array
+
+    """
+    ra = np.radians(ra)
+    dec = np.radians(dec)
+    roll = np.radians(roll)
+    ca = np.cos(ra)
+    sa = np.sin(ra)
+    cd = np.cos(dec)
+    sd = np.sin(dec)
+    cr = np.cos(roll)
+    sr = np.sin(roll)
+    # This is the transpose of the transformation matrix (related to
+    # translation of original perl code
+    rmat = np.array(
+        [[ca * cd,                  sa * cd,                sd     ],
+         [-ca * sd * sr - sa * cr, -sa * sd * sr + ca * cr, cd * sr],
+         [-ca * sd * cr + sa * sr, -sa * sd * cr - ca * sr, cd * cr]])
+
+    return rmat.transpose()
+
+
+def radec2eci(ra, dec):
+    """
+    Convert from RA,Dec to ECI.  The input ``ra`` and ``dec`` values can be 1-d
+    arrays of length N in which case the output ``ECI`` will be an array with
+    shape (3,N).
+
+    Borrowed from Ska.quatutil
+
+    :param ra: Right Ascension (degrees)
+    :param dec: Declination (degrees)
+    :returns: numpy array ECI (3-vector or 3xN array)
+    """
+    r = np.radians(ra)
+    d = np.radians(dec)
+    return np.array([np.cos(r) * np.cos(d), np.sin(r) * np.cos(d), np.sin(d)])
+
+
+#def roll_range(asp_roll):
```

Comment:
My intent here was to use the roll range to print something reasonable for how much one can expect ones events to deviate from real yag/zag just due to off-axis position and roll deviations (since we are using a single reference roll_nom).